go/vt/vttablet/tabletmanager/vreplication/vcopier_test.go

```diff
@@ -29,6 +29,103 @@ import (
 	"vitess.io/vitess/go/vt/vttablet/tabletserver/vstreamer"
 )
 
+func TestPlayerCopyTablesWithFK(t *testing.T) {
+	testForeignKeyQueries = true
+	defer func() {
+		testForeignKeyQueries = false
+	}()
+
+	defer deleteTablet(addTablet(100))
+
+	execStatements(t, []string{
+		"create table src2(id int, id2 int, primary key(id))",
+		"create table src1(id int, id2 int, primary key(id), foreign key (id2) references src2(id) on delete cascade)",
+		"insert into src2 values(1, 21), (2, 22)",
+		"insert into src1 values(1, 1), (2, 2)",
+		fmt.Sprintf("create table %s.dst2(id int, id2 int, primary key(id))", vrepldb),
+		fmt.Sprintf("create table %s.dst1(id int, id2 int, primary key(id), foreign key (id2) references dst2(id) on delete cascade)", vrepldb),
+	})
+	defer execStatements(t, []string{
+		"drop table src1",
+		fmt.Sprintf("drop table %s.dst1", vrepldb),
+		"drop table src2",
+		fmt.Sprintf("drop table %s.dst2", vrepldb),
+	})
+	env.SchemaEngine.Reload(context.Background())
+
+	filter := &binlogdatapb.Filter{
+		Rules: []*binlogdatapb.Rule{{
+			Match:  "dst1",
+			Filter: "select * from src1",
+		}, {
+			Match:  "dst2",
+			Filter: "select * from src2",
+		}},
+	}
+
+	bls := &binlogdatapb.BinlogSource{
+		Keyspace: env.KeyspaceName,
+		Shard:    env.ShardName,
+		Filter:   filter,
+		OnDdl:    binlogdatapb.OnDDLAction_IGNORE,
+	}
+	query := binlogplayer.CreateVReplicationState("test", bls, "", binlogplayer.VReplicationInit, playerEngine.dbName)
+	qr, err := playerEngine.Exec(query)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
```

Comment:
why is this happening in a defer? we should make the tests as simple as possible, and it's not clear why this has to happen in a defer instead of just after the other tests